src/databricks/labs/ucx/install.py

```diff
@@ -819,23 +828,37 @@ def _job_clusters(self, names: set[str]):
             )
         return clusters
 
-    def _cluster_node_type(self, spec: compute.ClusterSpec) -> compute.ClusterSpec:
-        cfg = self.current_config
-        valid_node_type = False
-        if cfg.custom_cluster_policy_id is not None:
-            if self._check_policy_has_instance_pool(cfg.custom_cluster_policy_id):
-                valid_node_type = True
-        if not valid_node_type:
-            if cfg.instance_pool_id is not None:
-                return replace(spec, instance_pool_id=cfg.instance_pool_id)
-            spec = replace(spec, node_type_id=self._ws.clusters.select_node_type(local_disk=True))
+    def _cluster_policy_definition(self, conf: dict, instance_profile: str) -> str:
+        policy_definition = {
+            "spark_version": {"type": "fixed", "value": self._ws.clusters.select_spark_version(latest=True)},
+            "node_type_id": {"type": "fixed", "value": self._ws.clusters.select_node_type(local_disk=True)},
+        }
+        if conf:
```

Comment:
i think this `if` statement is redundant. it has to be there always, no?..

Reply:
if there is no conf info (when there is no external hms) then it will be None, removing if will give error on conf.items()